src/main/java/org/logstash/filters/DateFilter.java

```diff
@@ -0,0 +1,118 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.logstash.filters;
+
+import org.joda.time.Instant;
+import org.logstash.Event;
+import org.logstash.ext.JrubyEventExtLibrary.RubyEvent;
+import org.logstash.filters.parser.CasualISO8601Parser;
+import org.logstash.filters.parser.JodaParser;
+import org.logstash.filters.parser.TimestampParser;
+import org.logstash.filters.parser.TimestampParserFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DateFilter {
+  private final String sourceField;
+  private final String[] tagOnFailure;
+  private RubySuccessHandler successHandler;
+  private final List<ParserExecutor> executors = new ArrayList<>();
+  private final ResultSetter setter;
+
+  public interface RubySuccessHandler {
+    void success(RubyEvent event);
+  }
+
+  public DateFilter(String sourceField, String targetField, List<String> tagOnFailure, RubySuccessHandler successHandler) {
+    this.sourceField = sourceField;
+    this.tagOnFailure = tagOnFailure.toArray(new String[0]);
+    this.successHandler = successHandler;
+    if (targetField.equals("@timestamp")) {
+      this.setter = new TimestampSetter();
+    } else {
+      this.setter = new FieldSetter(targetField);
+    }
+  }
+
+  public DateFilter(String sourceField, String targetField, List<String> tagOnFailure) {
+    this.sourceField = sourceField;
+    this.tagOnFailure = tagOnFailure.toArray(new String[0]);
+    if (targetField.equals("@timestamp")) {
+      this.setter = new TimestampSetter();
+    } else {
+      this.setter = new FieldSetter(targetField);
+    }
+  }
+
+  public void acceptFilterConfig(String format, String locale, String timezone) {
+    TimestampParser parser = TimestampParserFactory.makeParser(format, locale, timezone);
+    if (parser instanceof JodaParser || parser instanceof CasualISO8601Parser) {
+      executors.add(new TextParserExecutor(parser, timezone));
+    } else {
+      executors.add(new NumericParserExecutor(parser));
+    }
+  }
+
+//  public void register() {
+//    // Nothing to do.
+//  }
+
+  //public Event[] receive(List<org.logstash.ext.JrubyEventExtLibrary.RubyEvent> rubyEvents) {
+
+  public List<RubyEvent> receive(List<RubyEvent> rubyEvents) {
+    for (RubyEvent rubyEvent : rubyEvents) {
+      Event event = rubyEvent.getEvent();
+      // XXX: Check for cast failures
+
+      ParseExecutionResult code = executeParsers(event);
+      if (ParseExecutionResult.NO_INPUT_FOUND == code) {
+        continue;
+      } else if (ParseExecutionResult.SUCCESS == code) {
+        if (successHandler != null) {
+          successHandler.success(rubyEvent);
+        }
+      } else {
+        for (String t : tagOnFailure) {
+          event.tag(t);
+        }
+      }
+    }
+    return rubyEvents;
+  }
+
+  public ParseExecutionResult executeParsers(Event event) {
+    Object input = event.getField(sourceField);
```

Comment:
When would this be a non-string object? Can we directly cast this to `String` type?